### DISCUSSION TITLE
Converting from OpenVPN access server to openvpn-ui

### DISCUSSION THREAD

**Discussion by jannoke** [d3vilh/openvpn-ui#70] [Q&A]
I'm starting to work on openvpn access server conversion as over the years it has gone ridiculously expensive. I kinda suprised that someone has not already paid for this as yearly cost can be around 10k for 100client setup.

openvpn-ui project seems quite fitting. 

on AS side I can see that client and server certificates are available in sqlite db. Also the routes (or nat) settings are. We have alot of IOT devices meaning that there are network behind vpn clients that needs to be accessed. Also there is some secureity questions - not all client should be able to access all networks connected to vpn. For now i guess i can filter out the routing parameters per client and generate them in staticclients files. There would still be question were clients could manually add a route and access network not intended for them. It would be fine for now.

So big question  - Maybe someone happen to do it already?

For now I would like some help on the workflow side  - What should I do to get client certs show up on openvpn-ui interface? Where is this list kept.

**Comment by d3vilh** [comment]
I hope you're considering migrating from the Community AS version.
Because folks from OpenVPN doing really great job to providing AS support services which are really great and let you have experience people behind your back for resolving your issues fast.

While this non-commercial UI project wasn't intended for big-scale business environments, it can still be a viable option if it fits your needs. With the support of the Community, we can achieve a lot together!

Although I haven't personally used Access Server, but as you described it, migrating user profiles shouldn't be a big challenge with some Python or Bash scripting. If you decide to make the switch, please share your experience with us, and feel free to contribute any scripts to pushing it in separate branch (we will help with testing).

The OpenVPN team is doing an outstanding job supporting the community and developing non-Community versions of Access Server and OpenVPN Connect. While their pricing model may sometimes be unbalanced, it's important to remember that it's their livelihood. After all, in the end, the Community always has alternatives :)

**Reply by jannoke** [reply]
I'm migrating from commercial version. The support has not been good for me then i needed it 4-5 years back. It took days to resolve billing-licensing issues.  It took Also there has not been considerable bugfixes or new features for some years. As for vpn solution for small bussiness - openvpn just works and there is not much to "support". I have contacted support 2-3 times over 10 year timespan. it could be thing for hundreds or thousands of clients with cluster setup, but for 100-200clients it's still "small" and does not need any tweaking - it just works as it should. So I'm considering openvpn AS commercial for huge clients which need fast reaction and SLA, not small and medium sized clients.
It's just my 2 cents on 10 year relationship with them. They have their place but they have moved on to only higher tier clients. But let's not focus on that. 

I will gladly provide back all the scripts that will come out of this. I have idea how to make all the export files from AS server. This is not a problem for me.  That research has already been done. But I need now help how to add these client certificates to openvpn-ui. So how would I go about if i wanted to add existing "clientXXX.conf" to openvpn-ui? If that is understood then I can go on with developing the migration scripts.
